src/main/kotlin/com/exactpro/th2/common/message/ParsedMessageBuilder.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.common.message
+
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.grpc.MessageMetadata
+import com.exactpro.th2.common.grpc.NullValue
+import com.exactpro.th2.common.grpc.Value
+
+class ParsedMessageBuilder : MessageBuilder<Message>() {
+    private var messageType: String? = null
+    private var fields = mutableMapOf<String, Value>()
+
+    override fun toProto(parentEventId: EventID): Message {
+        val metadataBuilder = MessageMetadata.newBuilder()
+            .setId(getMessageId())
+        if (messageType != null) {
+            metadataBuilder.messageType = messageType
+        }
+        if (properties != null) {
+            metadataBuilder.putAllProperties(properties)
+        }
+        if (protocol != null) {
+            metadataBuilder.protocol = protocol
+        }
+        val timestamp = getTimestamp()
+        if (timestamp != null) {
+            metadataBuilder.timestamp = timestamp
+        }
+        return Message.newBuilder()
+            .setParentEventId(parentEventId)
+            .setMetadata(metadataBuilder)
+            .putAllFields(fields)
+            .build()
+    }
+
+    fun addNullField(field: String): ParsedMessageBuilder {
+        this.fields[field] = Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build()
+        return this
+    }
+
+    fun addSimpleField(field: String, value: String): ParsedMessageBuilder {
+        this.fields[field] = Value.newBuilder().setSimpleValue(value).build()
+        return this
+    }
```

Comment:
We should cover four types of values:
simple, message
list of simple, list of message

Reply:
By common.proto` in th2-grpc-common it can be:
```
message Value {
  oneof kind {
    NullValue null_value = 1;
    string simple_value = 2;
    Message message_value = 3;
    ListValue list_value = 4;
  }
}
```
I added methods for `Message` and `ListValue`.

Reply:
`list[...]` doesn't seem to be much different from built-in `listOf(...)`, I would've prefer to use built-in collection builders where possible

Reply:
Do we combine both approaches in one builder?
For example 
java:
`var msg = factory.createMessage("session alias")
 msg.metadata() // getter 
      .sequence(3425) // setter
      .direction(FIRST) // setter
 msg.body() // getter
      .put("a", "value")
      .put("B", factory.createMessage()
          .body()
              .put("c", "value"))
      ...
`